File: TheaterSuggestions/CSharp/SeatsSuggestions/SuggestionMade.cs
using System.Collections.Generic;
using System.Linq;

namespace SeatsSuggestions;

/// <summary>
///     Occurs when a Suggestion is made.
/// </summary>
public class SuggestionMade
{
    private readonly List<Seat> _suggestedSeats;

    public SuggestionMade(SeatingOptionSuggested seatingOptionSuggested)
    {
        PartyRequested = seatingOptionSuggested.PartyRequested;
        PricingCategory = seatingOptionSuggested.PricingCategory;
        _suggestedSeats = seatingOptionSuggested.Seats;
    }

    public int PartyRequested { get; }
    public PricingCategory PricingCategory { get; }

    public IReadOnlyList<Seat> SuggestedSeats => _suggestedSeats;

    public IEnumerable<string> SeatNames()
    {
        return _suggestedSeats.Select(s => s.ToString());
    }

    public bool MatchExpectation()
    {
        return _suggestedSeats.Count == PartyRequested;
    }
}

File: TheaterSuggestions/TypeScript/SeatsSuggestions/src/suggestion-made.ts
import { PricingCategory } from "./pricing-category";
import { Seat } from "./seat";

/**
 * Occurs when a Suggestion is made.
 */
export class SuggestionMade {
  constructor(
    public readonly suggestedSeats: readonly Seat[],
    public readonly partyRequested: number,
    public readonly pricingCategory: PricingCategory
  ) {}

  seatNames = () => this.suggestedSeats.map(seat => seat.toString())

  matchExpectation = () => this.suggestedSeats.length === this.partyRequested

}
